From a53ff6b6571e2f14d85bacbc12ba2b6e74d76b6f Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Tue, 2 Dec 2025 12:33:17 +0100
Subject: [PATCH] linux: Increase the maximum number of open files

We are hitting the limit often, so let's increase the default limits.
---
 include/kernel-version.mk                     |  2 +-
 .../9999-increase-NR_OPEN-by-default.patch    | 61 +++++++++++++++++++
 2 files changed, 62 insertions(+), 1 deletion(-)
 create mode 100644 target/linux/generic/hack-6.6/9999-increase-NR_OPEN-by-default.patch

diff --git a/include/kernel-version.mk b/include/kernel-version.mk
index 958ed9be0c..b163849cb6 100644
--- a/include/kernel-version.mk
+++ b/include/kernel-version.mk
@@ -1,6 +1,6 @@
 
 # Use the default kernel version if the Makefile doesn't override it
-LINUX_RELEASE?=1
+LINUX_RELEASE?=2
 
 ifdef CONFIG_TESTING_KERNEL
   KERNEL_PATCHVER:=$(KERNEL_TESTING_PATCHVER)
diff --git a/target/linux/generic/hack-6.6/9999-increase-NR_OPEN-by-default.patch b/target/linux/generic/hack-6.6/9999-increase-NR_OPEN-by-default.patch
new file mode 100644
index 0000000000..76712cbb32
--- /dev/null
+++ b/target/linux/generic/hack-6.6/9999-increase-NR_OPEN-by-default.patch
@@ -0,0 +1,61 @@
+Index: linux-6.6.117/include/uapi/linux/fs.h
+===================================================================
+--- linux-6.6.117.orig/include/uapi/linux/fs.h
++++ linux-6.6.117/include/uapi/linux/fs.h
+@@ -34,8 +34,8 @@
+ 
+ /* Fixed constants first: */
+ #undef NR_OPEN
+-#define INR_OPEN_CUR 1024	/* Initial setting for nfile rlimits */
+-#define INR_OPEN_MAX 4096	/* Hard limit for nfile rlimits */
++#define INR_OPEN_CUR  8192	/* Initial setting for nfile rlimits */
++#define INR_OPEN_MAX 32768	/* Hard limit for nfile rlimits */
+ 
+ #define BLOCK_SIZE_BITS 10
+ #define BLOCK_SIZE (1<<BLOCK_SIZE_BITS)
+@@ -107,7 +107,7 @@ struct inodes_stat_t {
+ };
+
+
+-#define NR_FILE  8192	/* this can well be larger on a larger system */
++#define NR_FILE  32768	/* this can well be larger on a larger system */
+
+ /*
+  * Structure for FS_IOC_FSGETXATTR[A] and FS_IOC_FSSETXATTR.
+Index: linux-6.6.117/include/uapi/linux/limits.h
+===================================================================
+--- linux-6.6.117.orig/include/uapi/linux/limits.h
++++ linux-6.6.117/include/uapi/linux/limits.h
+@@ -2,7 +2,7 @@
+ #ifndef _UAPI_LINUX_LIMITS_H
+ #define _UAPI_LINUX_LIMITS_H
+ 
+-#define NR_OPEN	        1024
++#define NR_OPEN	        8192
+ 
+ #define NGROUPS_MAX    65536	/* supplemental group IDs are available */
+ #define ARG_MAX       131072	/* # bytes of args + environ for exec() */
+Index: linux-6.6.117/tools/include/uapi/linux/fs.h
+===================================================================
+--- linux-6.6.117.orig/tools/include/uapi/linux/fs.h
++++ linux-6.6.117/tools/include/uapi/linux/fs.h
+@@ -34,8 +34,8 @@
+ 
+ /* Fixed constants first: */
+ #undef NR_OPEN
+-#define INR_OPEN_CUR 1024	/* Initial setting for nfile rlimits */
+-#define INR_OPEN_MAX 4096	/* Hard limit for nfile rlimits */
++#define INR_OPEN_CUR  8192	/* Initial setting for nfile rlimits */
++#define INR_OPEN_MAX 32768	/* Hard limit for nfile rlimits */
+ 
+ #define BLOCK_SIZE_BITS 10
+ #define BLOCK_SIZE (1<<BLOCK_SIZE_BITS)
+@@ -107,7 +107,7 @@ struct inodes_stat_t {
+ };
+
+
+-#define NR_FILE  8192	/* this can well be larger on a larger system */
++#define NR_FILE  32768	/* this can well be larger on a larger system */
+
+ /*
+  * Structure for FS_IOC_FSGETXATTR[A] and FS_IOC_FSSETXATTR.
-- 
2.52.0

